src/App.tsx

```diff
@@ -1,14 +1,51 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
+// eslint-disable-next-line import/no-cycle
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getUser } from './api';
+import { Todo } from './types/Todo';
+
+export enum FilterBy {
+  ALL = 'all',
+  ACTIVE = 'active',
+  COMPLETED = 'completed',
+  QUERY = 'byQuery',
+}
 
 export const App: React.FC = () => {
+  const [todoIsLoaded, setTodoisLoaded] = useState(false);
+  const [currentUserId, setCurrentUserId] = useState(0);
+  const [currentTodo, setCurrentTodo] = useState<Todo>({
+    id: 0,
+    title: '',
+    completed: false,
+    userId: 0,
+  });
+  const [filterTodoBy, setFilterTodoBy] = useState(FilterBy.ALL);
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const filterCallback = () => {
+    switch (filterTodoBy) {
+      case FilterBy.ACTIVE:
+        return (todo: Todo) => !todo.completed;
+
+      case FilterBy.COMPLETED:
+        return (todo: Todo) => todo.completed;
+
+      case FilterBy.QUERY:
+        return (todo: Todo) => todo.title.includes(searchQuery);
```

Comment:
```suggestion
        return (todo: Todo) => todo.title.includes(searchQuery.toLowerCase());
```

Add toLowerCase() to ignore case sensitivity in the input field